e2e/_suites/kubernetes-autodiscover/features/heartbeat.feature

```diff
@@ -13,12 +13,19 @@ Scenario: Monitor pod availability using hints with named ports
 Scenario: Monitor service availability using hints
   Given "heartbeat" is running with "hints enabled for services"
    When "redis service" is deployed with "monitor annotations"
+    And "redis" is running
    Then "heartbeat" collects events with "kubernetes.service.name:redis"
-    And "heartbeat" collects events with "monitor.status:down"
-   When "redis" is running
     And "heartbeat" collects events with "monitor.status:up"
     And "heartbeat" does not collect events with "monitor.status:down" during "20s"
 
+# A service without backend pods should be reported as down.
+Scenario: Monitor service unavailability using hints
+  Given "heartbeat" is running with "hints enabled for services"
+   When "redis service" is deployed with "monitor annotations"
+   Then "heartbeat" collects events with "kubernetes.service.name:redis"
+    And "heartbeat" collects events with "monitor.status:down"
+    And "heartbeat" does not collect events with "monitor.status:up" during "20s"
```

Comment:
Thanks for this change!

I have another question: are those three conditions (Then > And > And) required all together? Or can they be split into different scenarios? It could be the case where the first condition fails, and then we won't receive feedback about the other two.

Reply:
Umm, I would prefer to leave them together, for the same scenario we expect all the conditions to happen.

It would also work if they were in three different scenarios, but it would look a bit weird to me to have three identical scenarios where only the expectations change.

Also, maybe we want to add support for more complex conditions, these checks would be more accurate if we could do this:
```
   Then "heartbeat" collects events with "kubernetes.service.name:redis and monitor.status:down"
    And "heartbeat" does not collect events with "kubernetes.service.name:redis and monitor.status:up" during "20s"
```

Reply:
Ok, let's keep it like this and we can refactor that in a follow-up PR.

As a reference, this is possible:
```gherkin
Scenario: Monitor service unavailability using hints
  Given "heartbeat" is running with "hints enabled for services"
   When "redis service" is deployed with "monitor annotations"
   Then "heartbeat" collects events with
   | key    | value    |
   | config | <config> |
    And "heartbeat" does not collect events with "monitor.status:up" during "20s"
Examples:
    | config                        | 
    | kubernetes.service.name:redis |
    | monitor.status:down           |
```
but it would require changing all steps using that method to accept a data table:

```go
func collectsEventsWith(config string, table *messages.PickleStepArgument_PickleTable) error {
	return godog.ErrPending
}
```